instructor/common.py

```diff
@@ -259,7 +259,8 @@ def assignments_edit_strains(request):
     pk = request.session['assignment_id']
     assignment = get_object_or_404(models.Assignment, id=pk)
     extra_fields = 0
-    if 'add' in request.POST:
+    strains = models.Strains.objects.filter(assignment=assignment)
+    if 'add' in request.POST or len(strains) < 1:
```

Comment:
`models.Strains.objects.filter(assignment=assignment)` returns a list. 
in python `if []` is `false`, so you can just write something like:
`if 'add' in request.POST or models.Strains.objects.filter(assignment=assignment):`

this has also th benefit that you will do the query to the DB only if  `'add'` is not in `request.POST`